tests/components/homematicip_cloud/test_device.py

```diff
@@ -240,3 +240,32 @@ async def test_hmip_reset_energy_counter_services(hass, default_mock_hap_factory
     )
     assert hmip_device.mock_calls[-1][0] == "reset_energy_counter"
     assert len(hmip_device._connection.mock_calls) == 4  # pylint: disable=W0212
+
+
+async def test_hmip_multi_area_device(hass, default_mock_hap_factory):
+    """Test multi area device. Check if devices are created and referenced."""
+    entity_id = "binary_sensor.wired_eingangsmodul_32_fach_channel5"
+    entity_name = "Wired Eingangsmodul – 32-fach Channel5"
+    device_model = "HmIPW-DRI32"
+    mock_hap = await default_mock_hap_factory.async_get_mock_hap(
+        test_devices=["Wired Eingangsmodul – 32-fach"]
+    )
+
+    ha_state, hmip_device = get_and_check_entity_basics(
+        hass, mock_hap, entity_id, entity_name, device_model
+    )
+    assert ha_state
+
+    # get the entity
+    entity_registry = await er.async_get_registry(hass)
+    entity = entity_registry.async_get(ha_state.entity_id)
+    assert entity
+
+    # get the device
+    device_registry = await dr.async_get_registry(hass)
+    device = device_registry.async_get(entity.device_id)
+    assert device.name == "Wired Eingangsmodul – 32-fach"
+
+    # get the hap
+    hap_device = device_registry.async_get(device.via_device_id)
+    assert hap_device.name == "Access Point"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)